encoding/v2/kzg/prover/prover.go

```diff
@@ -79,31 +80,67 @@ func NewProver(logger logging.Logger, kzgConfig *KzgConfig, encoderConfig *encod
 	return proverGroup, nil
 }
 
-func (e *Prover) GetFrames(data []byte, params encoding.EncodingParams) ([]*encoding.Frame, error) {
+func (e *Prover) GetFrames(data []byte, params encoding.EncodingParams) ([]*encoding.Frame, []uint32, error) {
 	symbols, err := rs.ToFrArray(data)
 	if err != nil {
-		return nil, fmt.Errorf("ToFrArray: %w", err)
+		return nil, nil, fmt.Errorf("ToFrArray: %w", err)
 	}
 
 	prover, err := e.GetKzgProver(params)
 	if err != nil {
-		return nil, fmt.Errorf("get kzg prover: %w", err)
+		return nil, nil, fmt.Errorf("get kzg prover: %w", err)
 	}
 
-	kzgFrames, _, err := prover.GetFrames(symbols)
-	if err != nil {
-		return nil, fmt.Errorf("get frames: %w", err)
+	type encodeCResult struct {
+		chunks   []rs.FrameCoeffs
+		indices  []uint32
+		duration time.Duration
+		err      error
 	}
-
-	chunks := make([]*encoding.Frame, len(kzgFrames))
-	for ind, frame := range kzgFrames {
-		chunks[ind] = &encoding.Frame{
-			Coeffs: frame.Coeffs,
-			Proof:  frame.Proof,
+	encodeChan := make(chan encodeCResult, 1)
+	go func() {
+		defer close(encodeChan)
+		encodeStart := time.Now()
+		frames, indices, err := e.encoder.Encode(symbols, params)
+		encodingDuration := time.Since(encodeStart)
+		encodeChan <- encodeCResult{
+			chunks:   frames,
+			indices:  indices,
+			duration: encodingDuration,
+			err:      err,
 		}
+	}()
+
+	getProofsStart := time.Now()
+	proofs, err := prover.GetProofs(symbols)
+	getProofsDuration := time.Since(getProofsStart)
+
+	// Wait for both chunks and frames to have finished generating
+	encodeRes := <-encodeChan
```

Comment:
nit: Prefer it to be consistent to above variable `encodeCResult` so use `encodeResult`

Reply:
[9003873](https://github.com/Layr-Labs/eigenda/pull/2198/commits/900387381db1d08b4d02ccbb6bd2d5d6403a6c03)